src/k-provider/ovp/provider.js

```diff
@@ -59,7 +60,9 @@ export default class OVPProvider extends BaseProvider<OVPProviderMediaInfoObject
         return this._dataLoader.fetchData().then(
           response => {
             try {
-              resolve(this._parseDataFromResponse(response));
+              const mediaConfig = this._parseDataFromResponse(response);
+              const finalMediaConfig = RegexActionHandler.handleRegexAction(mediaConfig, response);
```

Comment:
finalMediaConfig is returned before RegexActionHandler.handleRegexAction() finishes since it is asynchronous 

You should do something like 

`const mediaConfig = this._parseDataFromResponse(response); RegexActionHandler.handhleRegexAction(mediaConfig, response).then((finalMediaConfig) => resolve( finalMediaConfig))`

Reply:
Or just `const mediaConfig = this._parseDataFromResponse(response); 
RegexActionHandler.handhleRegexAction(mediaConfig, response).then(resolve)`